storm-server/src/main/java/org/apache/storm/daemon/supervisor/BasicContainer.java

```diff
@@ -665,8 +666,10 @@ protected String javaCmd(String cmd) {
         commandList.addAll(classPathParams);
         commandList.add(getWorkerMain(topoVersion));
         commandList.add(topologyId);
+        if (numaId != null) {
+            supervisorId += Constants.NUMA_ID_SEPARATOR + numaId;
```

Comment:
The mkLaunchCommand is in BasicContainer - is there a situation where it wouldn't be called?

Reply:
I mostly worried about the order. If it's not called yet but accessed somewhere else, the result can be unexpected. We shouldn't implicitly change the meaning of the attribute `supervisorId`

Reply:
I'm putting this back here - there could be/are multiple implementations for how a ResourceIsolation manager could implement NUMA pinning but the supervisor ID MUST contain the numa id in the current implementation. But this supervisor id is effectively only used when the worker is launched. I feel it is appropriate to keep it here so that it's explicitly used as opposed to a super constructor where it can be missed when future changes are maded

Reply:
This way is also fine.  Originally it was

```
if (numaId != null) {
           this.supervisorId += Constants.NUMA_ID_SEPARATOR + numaId;
}
```
and it changed `this.supervisorId`. 

Currently `this.supervisorId` is not changed, so it is fine.